### PR TITLE
[TK-05021] Check for chain rollback

### PR DESCRIPTION
**Pull request by freesig**
- Adds unit tests to metadata agent activity
- Store the source chain items as `(AgentPubKey, u32  //header seq, HeaderHash //of the chain item) -> ()` to misc meta
- Changes misc meta to use a prefix instead of an enum
- Updates `get_activity(ChainItemKey) -> Iter<TimmedHeaderHash>`
- Updates `register_activity(&Header)`
- Unit tests pass
- Adds higher level tests too sys validation for chain activity
- Can only visually check these work for now because we are only logging failed chains.

### CODE REVIEW

File: crates/holochain/src/core/state/metadata/keys.rs [hunk 1]
```diff
@@ -297,45 +263,95 @@ impl IntoIterator for &LinkMetaKey<'_> {
...
+                // Get the header hash
+                buf.extend(h.clone().into_inner());
+                buf.into()
+            }
+        }
+    }
+}
+
+// TODO: This is way to fragile there must be a better way
+// get from the k bytes to the chain item key
```
Comment (freesig) [inline crates/holochain/src/core/state/metadata/keys.rs:338]: This might need to be better

Comment (maackle) [reply crates/holochain/src/core/state/metadata/keys.rs:338]: Yeah can you just use some consts, so when we change the consts this changes too?

File: crates/holochain/src/core/sys_validate.rs [hunk 2]
```diff
@@ -234,13 +236,48 @@ pub async fn check_valid_if_dna(
...
+        DatabaseResult::Ok(vault_count + judged_count + pending_count)
+    })?;
+
+    // Ok or log warning
+    if count == 0 {
+        return Ok(());
+    } else {
+        let s = tracing::warn_span!("agent_activity");
+        let _g = s.enter();
+        tracing::warn!(
```
Comment (maackle) [inline crates/holochain/src/core/sys_validate.rs:272]:
```suggestion
        // TODO: implement real rollback detection once we know what that looks like
        tracing::error!(
```

**Review by maackle — APPROVED** [review]
Looks good! Smooth transition from enum to struct w/ generic prefix.

**Pull request merged**